src/components/Task/Input.js

```diff
@@ -0,0 +1,35 @@
+import { useState } from "react";
+import { replaceTaskAtIndex, findIndex } from "../helpers/helpers";
+import { useRecoilState } from "recoil";
+import { ListOfTasks } from "../../store/store";
+import PropTypes from "prop-types";
+
+export const Input = ({ text, item }) => {
+  const [tasksList, setTasksList] = useRecoilState(ListOfTasks);
```

Comment:
This is more about architectural decisions. IMO, `Input` shouldn't have access to the whole `taskList`. `Input` is just a control. Especially, that here you modify only 1 item. We can review the architecture later, but this is a thing to reconsider.